gluten-core/src/main/scala/io/glutenproject/GlutenPlugin.scala

```diff
@@ -190,7 +190,8 @@ private[glutenproject] class GlutenDriverPlugin extends DriverPlugin with Loggin
 
 private[glutenproject] class GlutenExecutorPlugin extends ExecutorPlugin {
   private var executorEndpoint: GlutenExecutorEndpoint = _
-  private val taskListeners: Seq[TaskListener] = Array(TaskResources)
+  private val taskListeners: Seq[TaskListener] =
+    (immutable.ArraySeq.unsafeWrapArray(Array(TaskResources)))
```

Comment:
`immutable.ArraySeq.unsafeWrapArray` is a Scala 2.13 only api, if this performance optimization is necessary, this file may need to be split into Scala 2.12 and Scala 2.13 versions